src/app/data-planes/components/DataPlaneList.vue

```diff
@@ -102,56 +109,33 @@
         {{ t('common.collection.none') }}
       </template>
     </template>
-    <template #certificate="{ row: item }">
-      {{
-        item.dataplaneInsight?.mTLS?.certificateExpirationTime ?
-          formatIsoDate(new Date(item.dataplaneInsight.mTLS.certificateExpirationTime).toUTCString()) :
-          t('data-planes.components.data-plane-list.certificate.none')
-      }}
-    </template>
```

Comment:
Why are we changing this in this PR?

Reply:
This shouldn’t been in the `transformToTableData` function from the start. No idea why it was added inline in the first place.

Reply:
> I think it should stay inline. It was probably done inline because as a general rule for the application we prefer to do these relatively simple things inline.

As long as a table has a `transformToTableDatas` that’s where the data transformations should happen. Not in two places. That’s just confusing.

> I think further down the line it will be these `transformToTableData`s that will be removed. We shouldn't need to perform lots of data reshaping here just to print some JSON data into a table.

This table is a really good example for why that’s not true. We do in fact need to do quite some data transformation to render the tables here.

Once we actually have established data transformation to happen on the data sourcing layer, this becomes a different conversation.

So, please don’t change this `transformToTableData` pattern where in use. It’s very useful. If you want to change this, please provide the data transformations on the data layer and specifically **not** inline.

Reply:
1. Could you tell me why this change is necessary for this PR?

> Once we actually have established data transformation to happen on the data sourcing layer, this becomes a different conversation.

We have, this is why we are iteratively moving everything out of data reshaping in .vue files and either inline or into the data layer. There are many reasons for doing this, one of them being we aren't locking this sort of data reshaping away in a single file so it can't be reused across multiple views. But I also think its interesting that in doing this we are also spotting inconsistencies in the application, as was the case with my recent zones PR.

Putting this sort of stuff in a reusable data layer means we don't have different implementations of the same data reshaping dotted around the codebase in unclear places (something I keep pointing out we need to avoid).

The problem we have here is I'm gradually working towards putting things inline (as we keep saying we should be doing) and moving anything else more complicated to a reusable data later, and you are flip-flopping them back.

---

Ideally we'd leave this as is, it has nothing to do with this PR. But if not could you at least put it in a different PR so we can make progress with this line of work without getting caught up in these sorts of discussions again.

Reply:
> The problem we have here is I'm gradually working towards putting things inline (as we keep saying we should be doing) and moving anything else more complicated to a reusable data later, and you are flip-flopping them back.

But we really shouldn’t be moving stuff inline _more_ 😬. We should do that less. **Especially** since you want to move things into the data layer you shouldn’t keep putting things inline when there is an existing function that encapsulates completely the table’s data transformation needs. That just makes it harder to move this away iteratively. Not easier. This provides you with an extremely straightforward thing to look at: is it in the transform function? Well then I’ll likely need to move it into the data source layer. By adding it in the template, you created a second place to remember.

I’m basically cleaning up what shouldn’t have gone in the template in the first place.

Reply:
We need to avoid discussing these things in PRs like this. The PR ideally change things within the scope of the PR and nothing more. Apart from that I'm kinda ok doing little clean up things as we go as long as they are small and not controversial.

This is obviously controversial, and we've had the same discussion many times before.

Please undo this change so we can merge this PR. If you want to discuss it separately make the change in another PR.

That means we can move forwards with this work.

---

Just to point out, whether we should be doing this inline or not, the fact of the matter is, this isn't within the scope of this PR. Please can you remove it and put it in another PR if you think we should change it.

If you have an argument that you think is relevant in moving away from keeping PRs within scope then please bring it up offline. As far as I'm aware keeping PRs within scope is an approach recommended and documented by the rest of the team.

Reply:
I undid this change now and will file it as a separate PR.

Reply:
Thankyou. Could you put it back in the same place so its not even in the change set please? I know you don't appreciate people unnecessarily changing the git history. I'd prefer it if there was no change here at all.

Reply:
...

Reply:
Done. Unsatisfied with anything else?

Reply:
Not right now no, I think we are good now, thank you for doing that. Let me take one last look at this incase, I think I can approve this now.